### PR TITLE
docs(readme): Revamp README

### PR DESCRIPTION
**Pull request by francoischalifour**
This is a proposal to improve our README to become more attractive to developers relying on READMEs.

As Algolia is going more mature, I thought it would be an opportunity to work on the image of InstantSearch to appear more mature as well. We're getting more exposure so I think it's important to take time to work on this.

This PR is to trigger discussion and to work on it together. Let me know if I'm taking any wrong directions.

## Main changes

- Change to newer logo
- Remove emojis in titles because they break hyperlinks (we can find a solution if we really need them)
- Add badges to bring more confidence
- Update getting started code snippet to ES2015
- Link to getting started CodeSandbox
- Add a clear "Documentation" section
- Add a "Demos" section to see what you can build with InstantSearch.js
- Add a "Playground" section that links to the template generated with [Create InstantSearch App](https://github.com/algolia/create-instantsearch-app)
- Reword the "Contributing" section
- Add "License" section

## Things to consider

- Need to update other flavors' README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,110 +1,158 @@
...
+- [Contributing](#contributing)
+- [License](#license)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## About InstantSearch.js ⚡️
+</details>
 
-> ⚡ Lightning-fast search for your apps
+## About InstantSearch.js
```
Comment (francoischalifour) [inline README.md]: I think we should replace this section with "Why" and explain the use cases.

File: README.md [hunk 2]
```diff
@@ -1,110 +1,158 @@
...
+* **Bug report**. Is something not working as expected? [Send a bug report](https://github.com/algolia/instantsearch.js/issues/new?template=Bug_report.md).
+* **Feature request**. Would you like to add something to the library? [Send a feature request](https://github.com/algolia/instantsearch.js/issues/new?template=Feature_request.md).
+* **Documentation**. Did you find a typo in the doc? Click on the `edit this page` button.
+* **Development**. If you don't know where to start, you can check the open issues that are [tagged easy](https://github.com/algolia/instantsearch.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22Difficulty%3A++++++%E2%9D%84%EF%B8%8F+easy%22), the [bugs](https://github.com/algolia/instantsearch.js/issues?q=is%3Aissue+is%3Aopen+label%3A%22%E2%9D%A4+Bug%22) or [chores](https://github.com/algolia/instantsearch.js/issues?q=is%3Aissue+is%3Aopen+label%3A%22%E2%9C%A8+Chore%22).
 
-To get started in your contribution, you just need to:
- - fork the project and clone it locally
- - install the dependencies `yarn`
- - run the dev mode `yarn run dev` and [open dev-novel](http://localhost:8080/)
+To get started contributing, you need to:
```
Comment (bobylito) [inline README.md]: To start contributing?

**Comment by bobylito** [comment]
We have a [video tutorial in the docs ](https://www.algolia.com/doc/tutorials/search-ui/instant-search/build-an-instant-search-results-page/instantsearchjs/#introduction). Could that be helpful?

**Comment by francoischalifour** [comment]
I added the "Why" section. It doesn't sound very sexy though.

**Comment by bobylito** [comment]
> I added the "Why" section. It doesn't sound very sexy though.

it's better than before :)

**Comment by francoischalifour** [comment]
Should we go with this README or you have more suggestions?

**Comment by bobylito** [comment]
I think so :)

**Comment by francoischalifour** [comment]
Why not add a "Hiring" section like Netflix did for [Polly.JS](https://github.com/Netflix/pollyjs#were-hiring)?

**Comment by bobylito** [comment]
> Why not add a "Hiring" section like Netflix did for Polly.JS?

That's a good call. Can you add a link to the open source job offer?

**Pull request merged**